modules/representation_management/app/models/representation_management/form_2122_base.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+module RepresentationManagement
+  class Form2122Base
+    include ActiveModel::Model
+
+    FOUR_DIGIT_NUMBER = /\A\d{4}\z/
+    FIVE_DIGIT_NUMBER = /\A\d{5}\z/
+    NINE_DIGIT_NUMBER = /\A\d{9}\z/
+    TEN_DIGIT_NUMBER = /\A\d{10}\z/
+
+    veteran_attrs = %i[
+      veteran_first_name veteran_middle_initial veteran_last_name
+      veteran_social_security_number
+      veteran_va_file_number
+      veteran_date_of_birth
+      veteran_address_line1
+      veteran_address_line2
+      veteran_city
+      veteran_country
+      veteran_state_code
+      veteran_zip_code
+      veteran_zip_code_suffix
+      veteran_phone
+      veteran_email
+      veteran_service_number
+      veteran_insurance_numbers
+    ]
+
+    claimant_attrs = %i[
+      claimant_first_name
+      claimant_middle_initial
+      claimant_last_name
+      claimant_date_of_birth
+      claimant_relationship
+      claimant_address_line1
+      claimant_address_line2
+      claimant_city
+      claimant_country
+      claimant_state_code
+      claimant_zip_code
+      claimant_zip_code_suffix
+      claimant_phone
+      claimant_email
+    ]
+
+    consent_attrs = %i[
+      record_consent
+      consent_address_change
+      consent_limits
+    ]
+
+    attr_accessor(*[veteran_attrs, claimant_attrs, consent_attrs].flatten)
+
+    validates :veteran_first_name, presence: true, length: { maximum: 12 }
```

Comment:
@jvcAdHoc could you share an example please

Reply:
For example, submitting a Power of Attorney request calls [validate_json_schema](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb#L49) (defined [here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/app/controllers/claims_api/v2/veterans/base.rb)) and that would validate against either the [2122](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/config/schemas/v2/2122.json) or [2122a ](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/config/schemas/v2/2122a.json)schema. The approach might be over engineering for just one controller.

Reply:
It there's a better pattern, we can review it, make note of it, and use it going forward. We're not going to make any changes to this at this time. It can be refactored later if necessary.

Reply:
Btw, thanks Jon